src/ObjectStatus/ObjectStatus.js

```diff
@@ -0,0 +1,89 @@
+import classnames from 'classnames';
+import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { OBJECT_STATUS_SIZES, OBJECT_STATUS_TYPES } from '../utils/constants';
+import 'fundamental-styles/dist/icon.css';
+import 'fundamental-styles/dist/object-status.css';
+
+/** *Object Status* is a short text that represents the semantic status of an object. It has a semantic color and an optional icon.
+ * Typically, the object status is used in the dynamic page header and as a status attribute of a line item in a table. */
+
+const ObjectStatus = React.forwardRef(({ ariaLabel, children, className, glyph, indication, inverted, link, onClick, size, status, ...props }, ref) => {
+    const objectStatusClasses = classnames(
+        'fd-object-status',
+        {
+            [`sap-icon--${glyph}`]: !!glyph,
+            [`fd-object-status--indication-${indication}`]: !!indication,
+            'fd-object-status--inverted': inverted,
+            'fd-object-status--large': size === 'l',
+            [`fd-object-status--${status}`]: !!status,
+            ['fd-object-status--link']: !!link || !!onClick
+        },
+        className
+    );
+    let StatusTag = 'span';
+    let semanticProps = {};
+
+    if (link?.length) {
+        StatusTag = 'a';
+        semanticProps.href = link;
+    } else if ( typeof onClick === 'function') {
+        StatusTag = 'button';
+        semanticProps.onClick = onClick;
+        semanticProps.style = {
+            background: 'transparent'
+        };
```

Comment:
I realized that this won't play well with `inverted` status. Either we should get the styles for `<button>` into fundamental-styles or use `<span>` with `role='button'`, `onClick` handler, `onKeyPress` handler, and `tabIndex=0`.